# Use Wolfi-compatible cp arguments and we don't need to check the release under /etc/redhat-release.
From 338a381453e2b6d2ade93b0a24022aa7b44a6a65 Mon Sep 17 00:00:00 2001
From: Dentrax <furkan.turkal@chainguard.dev>
Date: Wed, 20 Nov 2024 00:54:48 +0300
Subject: [PATCH] fix entrypoint for wolfi

Signed-off-by: Dentrax <furkan.turkal@chainguard.dev>
---
 connect-base/3.0/docker-entrypoint.sh | 12 +++---------
 1 file changed, 3 insertions(+), 9 deletions(-)

diff --git a/connect-base/3.0/docker-entrypoint.sh b/connect-base/3.0/docker-entrypoint.sh
index 45ffb3c..520711d 100755
--- a/connect-base/3.0/docker-entrypoint.sh
+++ b/connect-base/3.0/docker-entrypoint.sh
@@ -261,15 +261,9 @@ case $1 in
         echo "      OFFSET_FLUSH_TIMEOUT_MS=$CONNECT_OFFSET_FLUSH_TIMEOUT_MS"
         echo "      SHUTDOWN_TIMEOUT=$CONNECT_TASK_SHUTDOWN_GRACEFUL_TIMEOUT_MS"
 
-	# Choose the right `cp` argument, `--update=none` is not available on RHEL
-	release=`cat /etc/redhat-release | cut -d ' ' -f1`
-	if [ $release = "Fedora" ]; then
-	    cp_arg="-r --update=none"
-	else
-	    cp_arg="-rn"
-	fi
-	# Copy config files if not provided in volume
-        cp $cp_arg  $KAFKA_HOME/config.orig/* $KAFKA_HOME/config
+        # Patched by Chainguard
+        # Use Wolfi-compatible cp arguments.
+        cp -r -n  $KAFKA_HOME/config.orig/* $KAFKA_HOME/config
 
         #
         # Configure the log files ...
-- 
2.39.5 (Apple Git-154)

